models/db/context.go

```diff
@@ -183,6 +184,24 @@ func DeleteByBean(ctx context.Context, bean interface{}) (int64, error) {
 	return GetEngine(ctx).Delete(bean)
 }
 
+func DeleteByID(ctx context.Context, id int64, bean interface{}) (int64, error) {
+	return GetEngine(ctx).ID(id).NoAutoTime().Delete(bean)
+}
+
+func FindIDs(ctx context.Context, tableName, idCol string, cond builder.Cond) ([]int64, error) {
+	ids := make([]int64, 0, 10)
+	if err := GetEngine(ctx).Table(tableName).Cols(idCol).
+		Where(cond).Find(&ids); err != nil {
+		return nil, err
+	}
+	return ids, nil
+}
+
+func DecrByIDs(ctx context.Context, ids []int64, decrCol string, bean interface{}) error {
+	_, err := GetEngine(ctx).Decr(decrCol).In("id", ids).NoAutoCondition().NoAutoTime().Update(bean)
```

Comment:
What does `NoAutoCondition` do?
Does it ignore the non-default values of `bean`?

Reply:
`NoAutoCondition` means Xorm will not read `bean`'s field values as where conditions.